portal/worker/src/walrus-sites-sw.ts

```diff
@@ -50,8 +54,40 @@ self.addEventListener("fetch", async (event) => {
     console.log("Parsed URL: ", parsedUrl);
 
     if (requestDomain == portalDomain && parsedUrl && parsedUrl.subdomain) {
-        console.log("fetching from the service worker");
-        event.respondWith(resolveAndFetchPage(parsedUrl));
+        event.respondWith((async () => {
+            // Clean the cache of expired entries
+            await cleanExpiredCache();
```

Comment:
Remark: Here, we are incurring the cost of checking the whole list every time we fetch a resource. This is probably unnecessary.

Suggestion: Instead of checking the whole list
- check if there is a match in the cache for the requested site;
  - if there is a match, check its expiration:
    - if it's expired, remove the entry from the cache
    - if it's not expired, use the entry
  - if there is no match, fetch directly

this way, we still have the property that the entries we fetch from the cache are "fresh", but for an O(1) cost instead of O(n).

Reply:
(we may still want to periodically trigger a full cache clean)